core/src/main/scala/org/apache/spark/sql/SnappySession.scala

```diff
@@ -1855,7 +1855,11 @@ object SnappySession extends Logging {
         evaluatePlan(df, session, key.sqlText, key)
       }
     }
```

Comment:
Oops. Looks like I was wrong all the way. the performance penalty is only for the initialization and not for reading. Embarrassing. Will add it.